File: BinarySearch.java
public class BinarySearch {
    public static int binarySearch(int[] A , int size , int key){
        int hi = size ;
        int lo = 0;

        while (lo <= hi){
            int mid = (hi + lo)/2;
            if(key == A[mid]){
                return mid;
            }else if(key < A[mid]){
                hi = mid-1;
            }else{
                lo = mid + 1;
            }
        }

        return -1;
    }

    public static void main(String[] args) {
        int[] Arrray = {3 , 6 , 8 , 12 , 14 , 17 , 25 , 29 , 31 , 36 , 42 , 47 , 53 , 55 , 62 };
        if(binarySearch(Arrray , Arrray.length , 31 ) == -1){
            System.out.println("Not Found");
        }else{
            System.out.println("Found at position " + binarySearch(Arrray , Arrray.length , 31) );
        }
    }
}

File: BinarySearch.cpp
#include <bits/stdc++.h>
using namespace std;

int binarySearch(vector<int> &arr , int key){
    int high = arr.size() - 1;
    int low = 0;

    while(low <= high){
        int mid = (low + high)/2;

        if(arr[mid] == key) return mid;

        if(arr[mid] < key){
            low = mid+1;
        }else{
            high = mid-1;
        }
    }

    return -1;
}
int main(){


    vector<int> arr = {3 , 5 , 6 , 8 , 9 , 10 , 12 , 15 , 16 , 18};
    cout<< binarySearch(arr , 8) << endl;
    
    return 0;
}